src/main/java/org/spongepowered/api/map/color/MapPalette.java

```diff
@@ -0,0 +1,168 @@
+
+    public static final MapColor GRASS = DummyObjectProvider.createFor(MapColor.class, "GRASS");
+
+    public static final MapColor GRAY = DummyObjectProvider.createFor(MapColor.class, "GRAY");
+
+    public static final MapColor GREEN = DummyObjectProvider.createFor(MapColor.class, "GREEN");
+
+    public static final MapColor ICE = DummyObjectProvider.createFor(MapColor.class, "ICE");
+
+    public static final MapColor IRON = DummyObjectProvider.createFor(MapColor.class, "IRON");
+
+    public static final MapColor LAPIS = DummyObjectProvider.createFor(MapColor.class, "LAPIS");
+
+    public static final MapColor LIGHT_BLUE = DummyObjectProvider.createFor(MapColor.class, "LIGHT_BLUE");
+
+    public static final MapColor LIME = DummyObjectProvider.createFor(MapColor.class, "LIME");
+
+    public static final MapColor MAGENTA = DummyObjectProvider.createFor(MapColor.class, "MAGENTA");
+
+    public static final MapColor NETHERRACK = DummyObjectProvider.createFor(MapColor.class, "NETHERRACK");
+
+    public static final MapColor OBSIDIAN = DummyObjectProvider.createFor(MapColor.class, "OBSIDIAN");
+
+    public static final MapColor PINK = DummyObjectProvider.createFor(MapColor.class, "PINK");
+
+    public static final MapColor PURPLE = DummyObjectProvider.createFor(MapColor.class, "PURPLE");
+
+    public static final MapColor QUARTZ = DummyObjectProvider.createFor(MapColor.class, "QUARTZ");
+
+    public static final MapColor RED = DummyObjectProvider.createFor(MapColor.class, "RED");
+
+    public static final MapColor SAND = DummyObjectProvider.createFor(MapColor.class, "SAND");
+
+    public static final MapColor SILVER = DummyObjectProvider.createFor(MapColor.class, "SILVER");
+
+    public static final MapColor SNOW = DummyObjectProvider.createFor(MapColor.class, "SNOW");
+
+    public static final MapColor STONE = DummyObjectProvider.createFor(MapColor.class, "STONE");
+
+    public static final MapColor TNT = DummyObjectProvider.createFor(MapColor.class, "TNT");
+
+    public static final MapColor WATER = DummyObjectProvider.createFor(MapColor.class, "WATER");
+
+    public static final MapColor WOOD = DummyObjectProvider.createFor(MapColor.class, "WOOD");
+
+    public static final MapColor YELLOW = DummyObjectProvider.createFor(MapColor.class, "YELLOW");
+
+    // SORTFIELDS:OFF
+
+    /**
+     * Returns all the possible {@link MapColor}s available, including the
+     * shaded colors.
+     *
+     * @return All the map colors including shades
+     */
+    public static ImmutableCollection<MapColor> getAll() {
+        return factory.getAll();
+    }
+
+    /**
+     * Returns all the possible named {@link MapColor}s available, this is
+     * basically just all the static entries in {@link MapPalette} but in an
+     * {@link ImmutableCollection<MapColor>}.
+     *
+     * @return All the named map colors
+     */
+    public static ImmutableCollection<MapColor> getAllNamed() {
```

Comment:
Again, useless if `MapColor` is a `CatalogType` as they all are "named" by contract.